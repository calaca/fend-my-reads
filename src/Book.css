.book {
  width: 138px;
  height: auto;
  padding: 5px;
  margin-bottom: 30px;
}

.book-cover {
  box-shadow:0;
  transition: 0.4s ease-in-out;
}

.book .book-cover:hover {
  box-shadow: 0px 0px 15px 0px rgba(0, 0, 0, 0.4);
}

.book-title {
  color: #848484;
  font-size: 16px;
  font-weight: 400;
  margin: 0;
}

.book-authors {
  margin: 5px 0 0 0;
  color: rgba(132, 132, 132, 0.65);
  font-size: 14px;
}
