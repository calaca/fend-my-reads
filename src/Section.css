section {
  background-color: #f3f9fA;
  padding: 30px 30px 0 30px;
  width: 100%;
  height: auto;
  margin-bottom: 20px;
  display: flex;
  flex-flow: row wrap;
  justify-content: space-between;
}

.section-title {
  color: #00b0af;
  text-transform: uppercase;
  font-size: 26px;
  font-weight: 200;
  width: 100%;
}
