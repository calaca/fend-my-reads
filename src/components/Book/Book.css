.book {
  width: 138px;
  height: auto;
  padding: 5px;
  margin-bottom: 30px;
  cursor: pointer;
}

.book-ratings {
  margin-bottom: 10px;
}

.book-cover {
  filter: grayscale(100%);
  box-shadow:0;
  transition: 0.4s ease-in-out;
}

.book:hover .book-cover {
  filter: grayscale(0);
  box-shadow: 0px 0px 15px 0px rgba(0, 0, 0, 0.4);
}

.book:hover .book-title {
  color: #00b0af;
}

.book:hover .book-authors {
  color: rgba(0, 176, 175, 0.65);
}

.book-title {
  color: #848484;
  font-size: 16px;
  font-weight: 400;
  margin: 0;
  transition: 0.4s ease-in-out;
}

.book-authors {
  margin: 5px 0 0 0;
  color: rgba(132, 132, 132, 0.65);
  font-size: 14px;
  transition: 0.4s ease-in-out;
}

.select-book-shelf select {
  -moz-appearance: none;
  -webkit-appearance: none;
  appearance: none;
  text-indent: 0.1;
  text-overflow: '';
  padding: 5px;
  background-color: #85dEdd;
  color: #fff;
  border: 0;
  cursor: pointer;
  border-radius: 4px;
  width: 100%;
  margin-bottom: 10px;
  transition: 0.4s ease-in-out;
}

.select-book-shelf select:hover {
  background-color: #00b0af;
}

.select-book-shelf select option:checked,
.select-book-shelf select option:hover {
  box-shadow: 0 0 10px 100px #019696 inset;
}

.select-book-shelf select:focus > option:checked { 
  background: #019696 !important;
}
