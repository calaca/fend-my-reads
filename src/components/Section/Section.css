section {
  background-color: #f3f9fA;
  padding: 30px 30px 0 30px;
  width: 100%;
  height: auto;
  margin-bottom: 20px;
  display: flex;
  flex-flow: row wrap;
  justify-content: flex-start;
}

.section-title {
  color: #00b0af;
  text-transform: uppercase;
  font-size: 26px;
  font-weight: 200;
  width: 100%;
}
