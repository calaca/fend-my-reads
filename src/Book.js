import React, { Component } from 'react';
import './Book.css';

class Book extends Component {
  render() {
    const book = this.props.book;
    const shelves = this.props.shelves;

    return (
      <div className="book">
        <div className="select-book-shelf">
          <select value={book.shelf} onChange={(event) => this.props.onHandleChange(event, book)}>
            {
              shelves.map(shelf => (
                <option key={shelf.name} value={shelf.name}>{shelf.title}</option>
              ))
            }
          </select>
        </div>
        <img
          className="book-cover"
          src={book.imageLinks.thumbnail}
          alt={book.title}
          title={book.title}
        />
        <p className="book-title">{book.title}</p>
        {
          book['authors'].map(author => (
            <p key={author} className="book-authors">{author}</p>
          ))
        }
      </div>
    )
  }
}

export default Book;
