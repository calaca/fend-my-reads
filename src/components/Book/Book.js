import React from 'react';
import PropTypes from 'prop-types';
import ReactStars from 'react-stars'
import './Book.css';

const Book = (props) => {
  const book = props.book;
  const shelves = props.shelves;

  // Handling books with no author/cover image
  const authors = book.authors || ['No author info'];
  let cover;
  if (typeof book.imageLinks === 'undefined') {
    cover = 'http://via.placeholder.com/350x150';
  } else {
    cover = book.imageLinks.thumbnail;
  }

  return (
    <div className="book">
      <div className="select-book-shelf">
        <select value={book.shelf} onChange={(event) => props.onHandleChange(event, book)}>
          {
            shelves.map(shelf => (
              <option key={shelf.name} value={shelf.name}>{shelf.title}</option>
            ))
          }
        </select>
      </div>
      <div className="book-ratings">
        <ReactStars
          count={5}
          size={20}
          value={book.rating}
          color1="#9c9c9c"
          color2="#f1c40f"
          half={false}
          onChange={(value) => props.onStarClick(value, book)}
        />
      </div>
      {
        <img
        className="book-cover"
        src={cover}
        alt={book.title}
        title={book.title}
      /> }
      <p className="book-title">{book.title}</p>
      {
        authors.map(author => (
          <p key={author} className="book-authors">{author}</p>
        ))
      }
    </div>
  )
};

Book.propTypes = {
  book: PropTypes.object.isRequired,
  shelves: PropTypes.array.isRequired,
  onHandleChange: PropTypes.func.isRequired,
  onStarClick: PropTypes.func.isRequired
};

export default Book;
