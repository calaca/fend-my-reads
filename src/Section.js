import React, { Component } from 'react';
import Book from './Book';
import './Section.css';

class Section extends Component {
  render() {
    const books = this.props.books;

    return (
      <section>
        <h2 className="section-title">{this.props.text}</h2>
        {
          books.map(book => (
            <Book key={book.id} book={book} />
          ))
        }
      </section>
    )
  }
}

export default Section;
