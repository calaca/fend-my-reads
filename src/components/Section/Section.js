import React from 'react';
import PropTypes from 'prop-types';
import Book from '../Book/Book';
import './Section.css';

const Section = (props) => {
  const books = props.books;
  return (
    <section>
      <h2 className="section-title">{props.shelf}</h2>
      {
        books.map(book => (
          <Book shelves={props.shelves} key={book.id} book={book} onHandleChange={props.onHandleChange} onStarClick={props.onStarClick} />
        ))
      }
    </section>
  )
};

Section.propTypes = {
  books: PropTypes.array.isRequired,
  shelves: PropTypes.array.isRequired,
  shelf: PropTypes.string.isRequired,
  onHandleChange: PropTypes.func.isRequired,
  onStarClick: PropTypes.func.isRequired
};

export default Section;
