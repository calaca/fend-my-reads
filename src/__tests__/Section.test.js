import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount } from 'enzyme';
import Section from '../components/Section/Section';
import { body } from '../__mocks__/MockData';
import Book from '../components/Book/Book';

describe('<Section />', () => {
  let handleChange, starClick;
  const { books, shelves } = body;
  const shelf = shelves[1];

  beforeEach(() => {
    handleChange = jest.fn();
    starClick = jest.fn();
  });

  afterEach(() => {
    handleChange.mockClear();
    starClick.mockClear();
  });

  it('renders without crashing', () => {
    const wrapper = shallow(
      <Section shelves={shelves} shelf={shelf.title} key='currentlyReading'
        books={books.filter(book => (book.shelf === shelf.name))}
        onHandleChange={handleChange} onStarClick={starClick} />
    );
  });

  it('renders correctly', () => {
    const wrapper = shallow(
      <Section shelves={shelves} shelf={shelf.title} key='currentlyReading'
        books={books.filter(book => (book.shelf === shelf.name))}
        onHandleChange={handleChange} onStarClick={starClick} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the shelf name', () => {
    const wrapper = shallow(
      <Section shelves={shelves} shelf={shelf.title} key='currentlyReading'
        books={books.filter(book => (book.shelf === shelf.name))}
        onHandleChange={handleChange} onStarClick={starClick} />
    );
    expect(wrapper.find('.section-title')).toHaveLength(1);
  });

  it('renders at least one <Book />', () => {
    const wrapper = mount(
      <Section shelves={shelves} shelf={shelf.title} key='currentlyReading'
        books={books.filter(book => (book.shelf === shelf.name))}
        onHandleChange={handleChange} onStarClick={starClick} />
    );
    expect(wrapper.find(Book).length).toBeGreaterThanOrEqual(1);
  });
});
