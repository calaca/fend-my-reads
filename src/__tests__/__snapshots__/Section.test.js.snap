// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Section /> renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Section
    books={
        Array [
            Object {
              "allowAnonLogging": true,
              "authors": Array [
                "William E. Shotts, Jr.",
              ],
              "averageRating": 4,
              "canonicalVolumeLink": "https://market.android.com/details?id=book-nggnmAEACAAJ",
              "categories": Array [
                "COMPUTERS",
              ],
              "contentVersion": "1.2.2.0.preview.2",
              "description": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
              "id": "nggnmAEACAAJ",
              "imageLinks": Object {
                "smallThumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                "thumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
              },
              "industryIdentifiers": Array [
                Object {
                  "identifier": "9781593273897",
                  "type": "ISBN_13",
                },
                Object {
                  "identifier": "1593273894",
                  "type": "ISBN_10",
                },
              ],
              "infoLink": "https://play.google.com/store/books/details?id=nggnmAEACAAJ&source=gbs_api",
              "language": "en",
              "maturityRating": "NOT_MATURE",
              "pageCount": 480,
              "panelizationSummary": Object {
                "containsEpubBubbles": false,
                "containsImageBubbles": false,
              },
              "previewLink": "http://books.google.com/books?id=nggnmAEACAAJ&dq=linux&hl=&cd=3&source=gbs_api",
              "printType": "BOOK",
              "publishedDate": "2012",
              "publisher": "No Starch Press",
              "ratingsCount": 2,
              "readingModes": Object {
                "image": false,
                "text": true,
              },
              "shelf": "currentlyReading",
              "subtitle": "A Complete Introduction",
              "title": "The Linux Command Line",
            },
          ]
    }
    onHandleChange={[Function]}
    onStarClick={[Function]}
    shelf="currently reading"
    shelves={
        Array [
            Object {
              "name": "wantToRead",
              "title": "want to read",
            },
            Object {
              "name": "currentlyReading",
              "title": "currently reading",
            },
            Object {
              "name": "read",
              "title": "read",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2
          className="section-title"
>
          currently reading
</h2>,
        Array [
          <Book
            book={
                        Object {
                                    "allowAnonLogging": true,
                                    "authors": Array [
                                      "William E. Shotts, Jr.",
                                    ],
                                    "averageRating": 4,
                                    "canonicalVolumeLink": "https://market.android.com/details?id=book-nggnmAEACAAJ",
                                    "categories": Array [
                                      "COMPUTERS",
                                    ],
                                    "contentVersion": "1.2.2.0.preview.2",
                                    "description": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
                                    "id": "nggnmAEACAAJ",
                                    "imageLinks": Object {
                                      "smallThumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                                      "thumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
                                    },
                                    "industryIdentifiers": Array [
                                      Object {
                                        "identifier": "9781593273897",
                                        "type": "ISBN_13",
                                      },
                                      Object {
                                        "identifier": "1593273894",
                                        "type": "ISBN_10",
                                      },
                                    ],
                                    "infoLink": "https://play.google.com/store/books/details?id=nggnmAEACAAJ&source=gbs_api",
                                    "language": "en",
                                    "maturityRating": "NOT_MATURE",
                                    "pageCount": 480,
                                    "panelizationSummary": Object {
                                      "containsEpubBubbles": false,
                                      "containsImageBubbles": false,
                                    },
                                    "previewLink": "http://books.google.com/books?id=nggnmAEACAAJ&dq=linux&hl=&cd=3&source=gbs_api",
                                    "printType": "BOOK",
                                    "publishedDate": "2012",
                                    "publisher": "No Starch Press",
                                    "ratingsCount": 2,
                                    "readingModes": Object {
                                      "image": false,
                                      "text": true,
                                    },
                                    "shelf": "currentlyReading",
                                    "subtitle": "A Complete Introduction",
                                    "title": "The Linux Command Line",
                                  }
            }
            onHandleChange={[Function]}
            onStarClick={[Function]}
            shelves={
                        Array [
                                    Object {
                                      "name": "wantToRead",
                                      "title": "want to read",
                                    },
                                    Object {
                                      "name": "currentlyReading",
                                      "title": "currently reading",
                                    },
                                    Object {
                                      "name": "read",
                                      "title": "read",
                                    },
                                  ]
            }
/>,
        ],
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "currently reading",
          "className": "section-title",
        },
        "ref": null,
        "rendered": "currently reading",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": "nggnmAEACAAJ",
        "nodeType": "function",
        "props": Object {
          "book": Object {
            "allowAnonLogging": true,
            "authors": Array [
              "William E. Shotts, Jr.",
            ],
            "averageRating": 4,
            "canonicalVolumeLink": "https://market.android.com/details?id=book-nggnmAEACAAJ",
            "categories": Array [
              "COMPUTERS",
            ],
            "contentVersion": "1.2.2.0.preview.2",
            "description": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
            "id": "nggnmAEACAAJ",
            "imageLinks": Object {
              "smallThumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
              "thumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
            },
            "industryIdentifiers": Array [
              Object {
                "identifier": "9781593273897",
                "type": "ISBN_13",
              },
              Object {
                "identifier": "1593273894",
                "type": "ISBN_10",
              },
            ],
            "infoLink": "https://play.google.com/store/books/details?id=nggnmAEACAAJ&source=gbs_api",
            "language": "en",
            "maturityRating": "NOT_MATURE",
            "pageCount": 480,
            "panelizationSummary": Object {
              "containsEpubBubbles": false,
              "containsImageBubbles": false,
            },
            "previewLink": "http://books.google.com/books?id=nggnmAEACAAJ&dq=linux&hl=&cd=3&source=gbs_api",
            "printType": "BOOK",
            "publishedDate": "2012",
            "publisher": "No Starch Press",
            "ratingsCount": 2,
            "readingModes": Object {
              "image": false,
              "text": true,
            },
            "shelf": "currentlyReading",
            "subtitle": "A Complete Introduction",
            "title": "The Linux Command Line",
          },
          "onHandleChange": [Function],
          "onStarClick": [Function],
          "shelves": Array [
            Object {
              "name": "wantToRead",
              "title": "want to read",
            },
            Object {
              "name": "currentlyReading",
              "title": "currently reading",
            },
            Object {
              "name": "read",
              "title": "read",
            },
          ],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2
            className="section-title"
>
            currently reading
</h2>,
          Array [
            <Book
              book={
                            Object {
                                          "allowAnonLogging": true,
                                          "authors": Array [
                                            "William E. Shotts, Jr.",
                                          ],
                                          "averageRating": 4,
                                          "canonicalVolumeLink": "https://market.android.com/details?id=book-nggnmAEACAAJ",
                                          "categories": Array [
                                            "COMPUTERS",
                                          ],
                                          "contentVersion": "1.2.2.0.preview.2",
                                          "description": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
                                          "id": "nggnmAEACAAJ",
                                          "imageLinks": Object {
                                            "smallThumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                                            "thumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
                                          },
                                          "industryIdentifiers": Array [
                                            Object {
                                              "identifier": "9781593273897",
                                              "type": "ISBN_13",
                                            },
                                            Object {
                                              "identifier": "1593273894",
                                              "type": "ISBN_10",
                                            },
                                          ],
                                          "infoLink": "https://play.google.com/store/books/details?id=nggnmAEACAAJ&source=gbs_api",
                                          "language": "en",
                                          "maturityRating": "NOT_MATURE",
                                          "pageCount": 480,
                                          "panelizationSummary": Object {
                                            "containsEpubBubbles": false,
                                            "containsImageBubbles": false,
                                          },
                                          "previewLink": "http://books.google.com/books?id=nggnmAEACAAJ&dq=linux&hl=&cd=3&source=gbs_api",
                                          "printType": "BOOK",
                                          "publishedDate": "2012",
                                          "publisher": "No Starch Press",
                                          "ratingsCount": 2,
                                          "readingModes": Object {
                                            "image": false,
                                            "text": true,
                                          },
                                          "shelf": "currentlyReading",
                                          "subtitle": "A Complete Introduction",
                                          "title": "The Linux Command Line",
                                        }
              }
              onHandleChange={[Function]}
              onStarClick={[Function]}
              shelves={
                            Array [
                                          Object {
                                            "name": "wantToRead",
                                            "title": "want to read",
                                          },
                                          Object {
                                            "name": "currentlyReading",
                                            "title": "currently reading",
                                          },
                                          Object {
                                            "name": "read",
                                            "title": "read",
                                          },
                                        ]
              }
/>,
          ],
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "currently reading",
            "className": "section-title",
          },
          "ref": null,
          "rendered": "currently reading",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": "nggnmAEACAAJ",
          "nodeType": "function",
          "props": Object {
            "book": Object {
              "allowAnonLogging": true,
              "authors": Array [
                "William E. Shotts, Jr.",
              ],
              "averageRating": 4,
              "canonicalVolumeLink": "https://market.android.com/details?id=book-nggnmAEACAAJ",
              "categories": Array [
                "COMPUTERS",
              ],
              "contentVersion": "1.2.2.0.preview.2",
              "description": "You've experienced the shiny, point-and-click surface of your Linux computer—now dive below and explore its depths with the power of the command line. The Linux Command Line takes you from your very first terminal keystrokes to writing full programs in Bash, the most popular Linux shell. Along the way you'll learn the timeless skills handed down by generations of gray-bearded, mouse-shunning gurus: file navigation, environment configuration, command chaining, pattern matching with regular expressions, and more. In addition to that practical knowledge, author William Shotts reveals the philosophy behind these tools and the rich heritage that your desktop Linux machine has inherited from Unix supercomputers of yore. As you make your way through the book's short, easily-digestible chapters, you'll learn how to: * Create and delete files, directories, and symlinks * Administer your system, including networking, package installation, and process management * Use standard input and output, redirection, and pipelines * Edit files with Vi, the world’s most popular text editor * Write shell scripts to automate common or boring tasks * Slice and dice text files with cut, paste, grep, patch, and sed Once you overcome your initial \\"shell shock,\\" you'll find that the command line is a natural and expressive way to communicate with your computer. Just don't be surprised if your mouse starts to gather dust. A featured resource in the Linux Foundation's \\"Evolution of a SysAdmin\\"",
              "id": "nggnmAEACAAJ",
              "imageLinks": Object {
                "smallThumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=5&source=gbs_api",
                "thumbnail": "http://books.google.com/books/content?id=nggnmAEACAAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
              },
              "industryIdentifiers": Array [
                Object {
                  "identifier": "9781593273897",
                  "type": "ISBN_13",
                },
                Object {
                  "identifier": "1593273894",
                  "type": "ISBN_10",
                },
              ],
              "infoLink": "https://play.google.com/store/books/details?id=nggnmAEACAAJ&source=gbs_api",
              "language": "en",
              "maturityRating": "NOT_MATURE",
              "pageCount": 480,
              "panelizationSummary": Object {
                "containsEpubBubbles": false,
                "containsImageBubbles": false,
              },
              "previewLink": "http://books.google.com/books?id=nggnmAEACAAJ&dq=linux&hl=&cd=3&source=gbs_api",
              "printType": "BOOK",
              "publishedDate": "2012",
              "publisher": "No Starch Press",
              "ratingsCount": 2,
              "readingModes": Object {
                "image": false,
                "text": true,
              },
              "shelf": "currentlyReading",
              "subtitle": "A Complete Introduction",
              "title": "The Linux Command Line",
            },
            "onHandleChange": [Function],
            "onStarClick": [Function],
            "shelves": Array [
              Object {
                "name": "wantToRead",
                "title": "want to read",
              },
              Object {
                "name": "currentlyReading",
                "title": "currently reading",
              },
              Object {
                "name": "read",
                "title": "read",
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
